src/infra/cluster.rs

```diff
@@ -282,62 +282,38 @@ pub async fn leave() -> Result<()> {
     Ok(())
 }
 
-#[inline(always)]
-pub fn get_cached_online_nodes() -> Option<Vec<Node>> {
-    if NODES.len() == 0 {
+fn get_cached_nodes(cond: fn(&Node) -> bool) -> Option<Vec<Node>> {
+    if NODES.is_empty() {
         return None;
     }
-    let mut node_list = Vec::new();
-    for node in NODES.iter() {
-        if node.status.eq(&NodeStatus::Online) {
-            node_list.push(node.value().clone());
-        }
-    }
-    Some(node_list)
+    Some(
+        NODES
+            .iter()
+            .filter_map(|node| cond(&node).then(|| node.clone()))
+            .collect(),
+    )
 }
 
 #[inline(always)]
-pub fn get_cached_online_query_nodes() -> Option<Vec<Node>> {
```

Comment:
I moved this function down, under `get_cached_online_querier_nodes`.
Now the functions are ordered by `cond`ition's complexity.